cromwell-backend/src/main/scala/cromwell/backend/provider/local/LocalBackend.scala

```diff
@@ -0,0 +1,348 @@
+package cromwell.backend.provider.local
+
+import java.nio.file.{Files, Path, Paths}
+import java.util.regex.Pattern
+
+import akka.actor.{ActorRef, Props}
+import akka.util.Timeout
+import better.files._
+import com.typesafe.scalalogging.StrictLogging
+import cromwell.backend.BackendActor
+import cromwell.backend.model._
+import cromwell.backend.provider.local.FileExtensions._
+import wdl4s.WdlExpression
+import wdl4s.types.{WdlFileType, WdlType}
+import wdl4s.values.{WdlFile, WdlSingleFile, WdlValue}
+
+import scala.annotation.tailrec
+import scala.collection.mutable.ArrayBuffer
+import scala.concurrent.duration._
+import scala.language.postfixOps
+import scala.sys.process._
+import scala.util.{Random, Try}
+
+object LocalBackend {
+  // Folders
+  val CromwellExecutionDir = "cromwell-executions"
+
+  // Files
+  val ReturnCodeFile = "rc"
+  val StdoutFile = "stdout"
+  val StderrFile = "stderr"
+  val ScriptFile = "script"
+
+  // Other
+  val DockerFlag = "docker"
+
+  def props(task: TaskDescriptor): Props = Props(new LocalBackend(task))
+}
+
+/**
+  * Executes a task in local computer through command line. It can be also executed in a Docker container.
+  * @param task Task descriptor.
+  */
+class LocalBackend(task: TaskDescriptor) extends BackendActor with StrictLogging {
+
```

Comment:
I still think we should automate this and then if we want to review it, we can do it.
I'm talking about spaces and formatting. Then we can open another discussion to talk about static code analysis.